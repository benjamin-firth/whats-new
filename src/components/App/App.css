* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

button {
  height: 25px;
}

body {
  background: #141527;
}

.app {
  display: flex;
  flex-wrap: wrap;
  height: 100vh;
  /* justify-content: center; */
  text-align: center;
  width: 100%;
}
