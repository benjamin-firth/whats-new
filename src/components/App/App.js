import React, { Component } from 'react';
import local from '../../data/local';
import entertainment from '../../data/entertainment';
import health from '../../data/health';
import science from '../../data/science';
import technology from '../../data/technology';
import './App.css';
import NewsContainer from '../NewsContainer/NewsContainer';
import Menu from '../Menu/Menu';



class App extends Component {
  constructor() {
    super();
    this.state = {
      currentPage: 'local',
      local,
      entertainment,
      health,
      science,
      technology
    }
  }

  render () {
    return (
      <div className="app">
        <Menu />
        <button onClick={() => this.setState({ currentPage: 'science' })}>Science News!</button>
        <button onClick={() => this.setState({ currentPage: 'entertainment' })}>Entertainment News!</button>
        <button onClick={() => this.setState({ currentPage: 'health' })}>Health News!</button>
        <button onClick={() => this.setState({ currentPage: 'technology' })}>technology News!</button>
        <button onClick={() => this.setState({ currentPage: 'local' })}>Local News!</button>
        <NewsContainer props={this.state[this.state.currentPage]} />
      </div>
    );
  }
}

export default App;
