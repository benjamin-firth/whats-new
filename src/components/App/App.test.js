import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
  wrapper = shallow(<App />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when changeCurrentPage is called', () => {
    const mockEvent = {target: {id: 'local'}};
    const expected = 'local';

    wrapper.instance().changeCurrentPage(mockEvent);
    expect(wrapper.state('currentPage')).toEqual(expected);
  })

  it('should update state when searchHandler is invoked', () => {
    const wrapper = shallow(<App />);

    const mockState = {
      currentPage: 'local',
      allData: {
        local: [
          {
            id: 1,
            headline: "The Who postpones Denver Concert at the Pepsi Center",
            img: "https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg",
            description: "Classic-rock band The Who has postponed its Denver concert.",
            url: "https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/"
          },
          {
            id: 2,
            headline: "Giant Chicken Skeleton In Denver’s Central Library Stands For More Than Just Dinner",
            img: "https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376",
            description: "The 24-foot-tall chicken skeleton towers over the people who pass through the main hall of the Denver Central Library.",
            url: "https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/"
          }
        ]
      }
    };

    const expected = [{
      id: 2,
      headline: "Giant Chicken Skeleton In Denver’s Central Library Stands For More Than Just Dinner",
      img: "https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376",
      description: "The 24-foot-tall chicken skeleton towers over the people who pass through the main hall of the Denver Central Library.",
      url: "https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/"
    }]
    
    wrapper.setState(mockState);
    wrapper.instance().searchHandler('towers');
    expect(wrapper.state('allData').filteredData).toEqual(expected);
  })
})


