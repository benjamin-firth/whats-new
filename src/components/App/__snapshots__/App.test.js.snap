// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="app"
>
  <SearchForm
    searchHandler={[Function]}
  />
  <Menu
    clickHandler={[Function]}
  />
  <NewsContainer
    props={Array []}
  />
</div>
`;
